aiida/transports/plugins/ssh.py

```diff
@@ -1294,35 +1294,105 @@ def exec_command_wait(self, command, stdin=None, combine_stderr=False, bufsize=-
         :param bufsize: same meaning of paramiko.
 
         :return: a tuple with (return_value, stdout, stderr) where stdout and stderr
-            are strings.
+            are both bytes.
         """
+        import socket
+        import time
+
         ssh_stdin, stdout, stderr, channel = self._exec_command_internal(command, combine_stderr, bufsize=bufsize)
 
         if stdin is not None:
             if isinstance(stdin, str):
                 filelike_stdin = io.StringIO(stdin)
+            elif isinstance(stdin, bytes):
+                filelike_stdin = io.BytesIO(stdin)
             else:
+                # It seems both StringIO and BytesIO work correctly when doing ssh_stdin.write(line)?
+                # (The ChannelFile is opened with mode 'b', but until now it always has been a StringIO)
+                if not isinstance(stdin, (io.BufferedIOBase, io.TextIOBase)):
+                    raise ValueError('You can only pass strings, bytes, BytesIO or StringIO objects')
+
                 filelike_stdin = stdin
 
             try:
-                for line in filelike_stdin.readlines():
+                for line in filelike_stdin:
                     ssh_stdin.write(line)
             except AttributeError:
-                raise ValueError('stdin can only be either a string of a file-like object!')
+                raise ValueError('stdin can only be either a string, bytes or a file-like object!')
```

Comment:
This `AttributeError` is actually raising because of `ssh_stdin`, right? Why you redirect to `ValueError` blaming `stdin`? I'm also wondering how would one test this (because codecov is complaining): the problem being caught is `self._exec_command_internal` returning an object without a `write` method, correct?

Reply:
You are right - this was needed before, but now we are checking the instance of all possible cases otherwise we already raise before, so this check (try/except) is not needed anymore. I'll remove it.